src/hdt.rs

```diff
@@ -57,6 +59,85 @@ impl Hdt {
         Ok(hdt)
     }
 
+    /// Creates an immutable HDT instance containing the dictionary and triples from the Path.
+    /// Will utilize a custom cached TriplesBitmap file if exists or create one if it does not exist.
+    /// The file path must point to the beginning of the data of an HDT file as produced by hdt-cpp.
+    /// FourSectionDictionary with DictionarySectionPlainFrontCoding and SPO order is the only supported implementation.
+    /// The format is specified at <https://www.rdfhdt.org/hdt-binary-format/>, however there are some deviations.
+    /// The initial HDT specification at <http://www.w3.org/Submission/2011/03/> is outdated and not supported.
+    /// # Example
+    /// ```
+    /// let hdt = hdt::Hdt::new_from_file(std::path::Path::new("tests/resources/snikmeta.hdt")).unwrap();
+    /// ```
+    #[cfg(feature = "cache")]
+    pub fn new_from_file(f: &std::path::Path) -> Result<Self, Box<dyn Error>> {
+        use log::warn;
+
+        let source = std::fs::File::open(f)?;
+        let mut reader = std::io::BufReader::new(source);
+        ControlInfo::read(&mut reader).wrap_err("Failed to read HDT control info")?;
+        Header::read(&mut reader).wrap_err("Failed to read HDT header")?;
+        let unvalidated_dict = FourSectDict::read(&mut reader).wrap_err("Failed to read HDT dictionary")?;
+        let abs_path = std::fs::canonicalize(f)?;
+        let index_file = format!(
```

Comment:
I prefer if the variable name doesn't conflict with the type, here it's a string but called a file.